modules/kubernetes-addons/velero/main.tf

```diff
@@ -0,0 +1,105 @@
+locals {
+  name = "velero"
+
+  argocd_gitops_config = {
+    enable             = true
+    serviceAccountName = local.name
+  }
+}
+
+module "helm_addon" {
+  source = "../helm-addon"
+
+  helm_config = merge({
+    name        = local.name
+    description = "A Helm chart for velero"
+    chart       = local.name
+    version     = "2.29.6"
+    repository  = "https://vmware-tanzu.github.io/helm-charts/"
+    namespace   = local.name
+    values = [
+      <<-EOT
+      configuration:
+        provider: aws
+        backupStorageLocation:
+          bucket: ${var.backup_s3_bucket}
+
+      initContainers:
+        - name: velero-plugin-for-aws
+          image: velero/velero-plugin-for-aws:v1.4.1
+          imagePullPolicy: IfNotPresent
+          volumeMounts:
+            - mountPath: /target
+              name: plugins
+      EOT
```

Comment:
move this to `values.yaml`. Use the existing approach similar to other add-ons. It's much cleaner to have `values.yaml` and keep it similar to how Helm works

Reply:
I had a values.yaml file only in my pull request for this configuration.

Reply:
> Thanks for the PR @greengangsta . It looks great now. Few comments added that needs addressing.
> 
> Also, you have added an empty file `examples/stateful/variables.tf`. Do you need this?

 I had added this file to take in input some variables (tenant, environment, zone for the example infra provisioning), possibly this change was made by bryantbiggs.

Reply:
updated https://github.com/aws-ia/terraform-aws-eks-blueprints/pull/407/commits/6354998ecf15c55d1f7087f4d5f5d42e6aea06ac